### PR TITLE
Add secondary index regression test

### PR DESCRIPTION
**Pull request by KieranBreen**
Add Secondary Index regression test.

### CODE REVIEW

File: TurfRegressionTests/Common/UnindexedTreesCollection.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import Foundation
+
+import Turf
+
+final class UnindexedTreesCollection: Collection {
```
Comment (jordanhamill) [inline TurfRegressionTests/Common/UnindexedTreesCollection.swift]: What is it you intend to test with this collection?

File: TurfRegressionTests/SecondaryIndexingRegressionTests.swift [hunk 2]
```diff
@@ -1,9 +1,63 @@
+import XCTest
+
+class SecondaryIndexingRegressionTests: XCTestCase {
```
Comment (jordanhamill) [inline TurfRegressionTests/SecondaryIndexingRegressionTests.swift]: You might have to rebase/merge master into your branch. I've added Quick and Nimble to write these tests.

You should be able to follow the outline of the Basic regression tests.

File: TurfRegressionTests/SecondaryIndexingRegressionTests.swift [hunk 3]
```diff
@@ -1,9 +1,63 @@
...
+            let indexedTreesCollection = transaction.readOnly(self.collections.indexedTrees)            
+            let fetchedTrees = indexedTreesCollection.findValuesWhere(indexedTreesCollection.indexed.height.isGreaterThan(20))
+
+            XCTAssertNotNil(fetchedTrees)
+//            XCTAssert(fetchedTrees.count == 2)
+        }
+
+    }
+
+//    func test_unindexed_collection_cant_be_queried() throws {
```
Comment (jordanhamill) [inline TurfRegressionTests/SecondaryIndexingRegressionTests.swift]: This won't throw - the type system means it won't even compile!

**Comment by jordanhamill** [comment]
Thanks for this! Definitely need to add support for floats and doubles. I'll get that up asap.

File: TurfRegressionTests/Common/UnindexedTreesCollection.swift (hunk 1 above)
Comment (KieranBreen) [inline TurfRegressionTests/Common/UnindexedTreesCollection.swift]: Ha, good question - I was hoping to test somehow that we don't even have the same methods, or that accessing the `indexed` property would either be `nil` or would throw an error.

I then kinda realised that Swift probably means I won't be able to (I was thinking along the lines of calling 'performSelector' or similar, no dice)

You think its OK to just have an indexed collection and call some of the Query related methods?

File: TurfRegressionTests/SecondaryIndexingRegressionTests.swift (hunk 2 above)
Comment (KieranBreen) [inline TurfRegressionTests/SecondaryIndexingRegressionTests.swift]: 👍

**Comment by KieranBreen** [comment]
Id like to pair with you on the Float/Double stuff maybe?

File: TurfRegressionTests/Common/UnindexedTreesCollection.swift (hunk 1 above)
Comment (jordanhamill) [inline TurfRegressionTests/Common/UnindexedTreesCollection.swift]: Yeah I think you'll find you can test the negative here. Testing the secondary index methods is enough for a regression test.

**Comment by jordanhamill** [comment]
Yeah thats cool. It shouldnt take more than a 30 mins to implement it.

File: TurfRegressionTests/Common/UnindexedTreesCollection.swift (hunk 1 above)
Comment (KieranBreen) [inline TurfRegressionTests/Common/UnindexedTreesCollection.swift]: 👍

File: Turf/Extensions/SecondaryIndex/Internals/WhereClauses.swift [hunk 4]
```diff
@@ -79,7 +79,7 @@ internal struct WhereClauses {
     }
 
     static func greaterThanOrEqual<Value: SQLiteType>(name name: String, value: Value) -> WhereClause {
-        return WhereClause(sql: "\(name) > ?",
+        return WhereClause(sql: "\(name) >= ?",
```
Comment (KieranBreen) [inline Turf/Extensions/SecondaryIndex/Internals/WhereClauses.swift:82]: Yay testing!

**Comment by KieranBreen** [comment]
@jordanhamill - should be good for final review.

File: Turf/Extensions/SecondaryIndex/Internals/WhereClauses.swift (hunk 4 above)
Comment (jordanhamill) [inline Turf/Extensions/SecondaryIndex/Internals/WhereClauses.swift:82]: 🌟

**Comment by jordanhamill** [comment]
This is a great first pass 👍 Thanks for doing this @KieranBreen

**Pull request merged**